### DISCUSSION TITLE
Website - Realm

### DISCUSSION THREAD

**Discussion by agkoutis** [simpleidserver/SimpleIdServer#607] [Ideas]
Hello,

While SID provides IdServerBuilder UseRealm() optional function, the Web Site does not provide similar functionality for the realm to be optional because of AuthMethodEffects http client url.

The result is that SID reponses in .well-known/openid-configuration the endpoints without using realm but the httpclient in WebSite builds url with the default realm.

Example  RequestUri = new Uri($"{_options.IdServerBaseUrl}/{realm}/authmethods")
Even if I replace the get Realm function there is still an unwanted slash between the " {_options.IdServerBaseUrl}/" and |/authmethods"|

Please consider providing a similar function as the SID UseRealm()  to the website.

Thanks

**Comment by simpleidserver** [comment]
Hello,

Indeed it must be possible to enable or disable realm in the administration website :)
I create a ticket #609 to support this feature ! It will be included in the release 4.0.5

KR,

SID

**Comment by simpleidserver** [comment]
Hello @agkoutis ,

I have made some modifications in the branch `release/4.0.5`.

You can disable Realm usage by following these steps:

1. Open the IdentityServer project and edit the `appsettings.json` file.
2. Set the `IsRealmEnabled` property to `false` and save the file.
3. Open the IdentityServer website and edit the `appsettings.json` file.
3. Set the `IsRealmEnabled` property to `false`, remove the master realm from the `DefaultSecurityOptions.Issuer` property, and save the file.
